ndonnx/_funcs.py

```diff
@@ -383,31 +387,12 @@ def equal(x, y):
     return _binary("equal", x, y)
 
 
-def _binary(func_name, x, y):
-    x_dtype = asarray(x).dtype
-    y_dtype = asarray(y).dtype
-    if (out := getattr(x_dtype._ops, func_name)(x, y)) is not NotImplemented:
-        return out
-    if (out := getattr(y_dtype._ops, func_name)(x, y)) is not NotImplemented:
-        return out
-    raise TypeError(
-        f"Unsupported operand type(s) for {func_name}: '{x_dtype}' and '{y_dtype}'"
-    )
-
-
-def _unary(func_name, x):
-    x = asarray(x)
-    if (out := getattr(x.dtype._ops, func_name)(x)) is not NotImplemented:
-        return out
-    raise TypeError(f"Unsupported operand type for {func_name}: '{x.dtype}'")
-
-
 def exp(x):
     return _unary("exp", x)
 
 
 def expm1(x):
-    return subtract(exp(x), 1)
+    return _unary("expm1", x)
```

Comment:
The `OperationsBlock` `expm1` and all(?) other operations here. If it is not implemented it will/should return `NotImplemented`. Why do we still have the need to deal with stringy function names here?

Reply:
Sorry I don't fully follow the question. `_unary` will dispatch the operation and if it returns `NotImplemented`, it will raise the `UnsupportedOperationError`.

Reply:
Sorry, I don't know what happened to that original comment of mine. Please see the edit. I am wondering why we have to resort to passing stringy function names.

Reply:
Okay I understand the question now. I think something like this existed a while ago and so it's stuck around - I can only imagine it's because the public API is untyped so it was a way to be defensive at some point.